jina/executors/indexers/vector.py

```diff
@@ -149,9 +149,11 @@ def _delete(self, keys):
                 self._size -= 1
 
     def delete(self, keys: Sequence[int], *args, **kwargs) -> None:
-        if kwargs.get('keys_precomputed') is not True:
-            keys = self._filter_nonexistent_keys(keys, self.ext2int_id.keys(), self.save_abspath)
-        self._delete(keys)
+        # if we dont check here, it crashes on `self.ext2int_id.keys()`
+        if self.size > 0:
+            if kwargs.get('keys_precomputed') is not True:
```

Comment:
please, use private `_delete` method and do not read from `kwargs` as it is done here, I thought this was removed from the code